javascript/app-discover/src/components/MenuBar.hooks.tsx

```diff
@@ -207,26 +176,99 @@ export function useFixedInterventionRangesToggleMenuItem() {
 	)
 }
 
-export function useTogglePauseButtonMenuItem() {
-	const [pauseAutoRun, setPauseAutoRun] = useRecoilState(PauseAutoRunState)
-	const paused = pauseAutoRun === CausalDiscoveryAlgorithm.None
-	const togglePaused = useCallback(() => {
-		setPauseAutoRun(paused ? undefined : CausalDiscoveryAlgorithm.None)
-	}, [setPauseAutoRun, paused])
+export function useAutoLayoutToggleMenuItem() {
+	const [autoLayoutEnabled, setAutoLayoutEnabled] = useRecoilState(
+		AutoLayoutEnabledState,
+	)
+	const handleClick = useRecoilCallback(
+		({ snapshot, set }) =>
+			async () => {
+				set(AutoLayoutEnabledState, true)
+				const autoLayoutSnapshot = await snapshot.getPromise(
+					AutoLayoutEnabledState,
+				)
+				if (autoLayoutSnapshot) {
+					setTimeout(() => {
+						set(AutoLayoutEnabledState, false)
+					}, 10)
+				}
+			},
+		[],
+	)
+
+	const menuProps = useMemo(
+		() => ({
+			items: [
+				{
+					key: 'autoLayout',
+					text: 'Auto-layout',
+					onClick: () => setAutoLayoutEnabled(true),
+				},
+			],
+		}),
+		[setAutoLayoutEnabled],
+	)
 
 	return useMemo(
 		() => ({
-			key: 'pause-toggle-button',
+			key: 'auto-layout-button',
+			onRender: () => (
+				<ButtonWrapper>
+					<DefaultButton
+						split
+						menuProps={menuProps}
+						checked={autoLayoutEnabled}
+						onClick={() => void handleClick()}
+						text={autoLayoutEnabled ? 'Auto-layout' : 'Layout'}
+					/>
+				</ButtonWrapper>
+			),
+		}),
+		[handleClick, autoLayoutEnabled, menuProps],
+	)
+}
+
+export function useRunButtonMenuItem() {
+	const [autoRun, setAutoRun] = useRecoilState(AutoRunState)
+	const { run, isLoading, stop } = useCausalDiscoveryRunner()
+	const isRunning = useMemo(
+		(): boolean => isLoading || autoRun,
+		[isLoading, autoRun],
+	)
+
+	const handleClick = useCallback(() => {
+		return void (isRunning ? stop() : run())
```

Comment:
the void here will ignores any promise returned from stop() or run(), how are errors being handled? Should we add a .catch() here instead of the void?

Reply:
each stop and run functions have its catch to log the possible errors